libs/core/datastructures/include/hpx/datastructures/detail/small_vector.hpp

```diff
@@ -536,6 +548,8 @@ namespace hpx::detail {
         // precondition: all uninitialized
         void do_move_assign(small_vector&& other) noexcept
         {
+            // We assume that the template parameters of "other" are the same
+            // as "this".
```

Comment:
The comment is superfluous: the _injected-class-name_ `small_vector` on line 549 literally means `small_vector<T, MinInlineCapacity, emulate_inplace_vector>` because that's how the language works.